src/colony/lua_tm.c

```diff
@@ -427,9 +432,13 @@ static int l_tm_ssl_read (lua_State* L)
   uint8_t buf[20000];
   size_t buf_len = sizeof(buf);
   int err = tm_ssl_read(session, buf, &buf_len);
-  (void) err;
+  if (err < 0) {
```

Comment:
In this case, the errors *will* be negative: https://github.com/tessel/runtime/blob/master/src/tm_ssl.c#L38

However, I would consider that a coincidence/internal detail and simply treat any non-zero value as an error for consistency.